### DISCUSSION TITLE
Event for selecting text?

### DISCUSSION THREAD

**Discussion by ed2050** [zauberzeug/nicegui#4100] [Q&A] [answered]
### Question

How would you detect when user selects text on the page?  Say user highlights text with mouse, now nicegui needs to do something with it.

If user triggers another action (keypress, button, etc) then it's fairly straightforward.  Just call `text = await ui.run_javascript (window.getSelection().toString())`.  A pain but doable.

But what if user doesn't perform another action?  I don't see any events in [nicegui](https://nicegui.io/documentation/section_action_events) or [MDN](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement#events) to detect it.

BTW - urls to MDN and Quasar [here](https://nicegui.io/documentation/section_action_events#generic_events) should be clickable links.  Right now just plain text.

**Comment by rodja** [comment, marked as answer]
There is the `selectionchanged` event: https://developer.mozilla.org/en-US/docs/Web/API/Document/selectionchange_event

I made the links clickable on main. Will be updated with the upcoming release.

**Comment by ed2050** [comment]
Thanks.  Nicegui docs only links to [element events](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement#events).  Should add a link to [document events](https://developer.mozilla.org/en-US/docs/Web/API/Document#events) too.
